scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,209 @@
+
+
+    Extend the original example to a sphere and plot interpolation
+    points in 3D:
+
+    >>> from mpl_toolkits.mplot3d import proj3d
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111, projection='3d')
+
+    Plot the unit sphere for reference (optional):
+
+    >>> u = np.linspace(0, 2 * np.pi, 100)
+    >>> v = np.linspace(0, np.pi, 100)
+    >>> x = np.outer(np.cos(u), np.sin(v))
+    >>> y = np.outer(np.sin(u), np.sin(v))
+    >>> z = np.outer(np.ones(np.size(u)), np.cos(v))
+    >>> ax.plot_surface(x, y, z, color='y', alpha=0.1)
+
+    Interpolating over a larger number of points
+    may provide the appearance of a smooth curve on
+    the surface of the sphere, which is also useful
+    for discretized integration calculations on a
+    sphere surface:
+
+    >>> start = np.array([1, 0, 0])
+    >>> end = np.array([0, 0, 1])
+    >>> t_vals = np.linspace(0, 1, 200)
+    >>> result = geometric_slerp(start,
+    ...                          end,
+    ...                          t_vals)
+    >>> ax.plot(result[...,0],
+    ...         result[...,1],
+    ...         result[...,2],
+    ...         c='k')
+    >>> plt.show()
+    """
+
+    start = np.asanyarray(start)
+    end = np.asanyarray(end)
+
+    if start.ndim != 1 or end.ndim != 1:
+        raise ValueError("Start and end coordinates "
+                         "must be one-dimensional")
+
+    if start.size != end.size:
+        raise ValueError("The dimensions of start and "
+                         "end must match (have same size)")
+
+    if start.size == 0 or end.size == 0:
+        raise ValueError("One or both input coordinates are empty")
+
+    if np.array_equal(start, end):
+        raise ValueError("Start and end coordinates cannot be the same")
+
+    if start.size == 1 or end.size == 1:
+        raise ValueError("Cannot interpolate the 0-sphere")
+
+    if not isinstance(tol, float):
+        raise ValueError("tol must be a float")
+    else:
+        tol = math.fabs(tol)
+
+    coord_dist = euclidean(start, end)
+
+    # diameter of 2 within tolerance means antipodes, which is a problem
+    # for all unit n-spheres (even the 0-sphere would have an ambiguous path)
+    if np.allclose(coord_dist, 2.0, rtol=0, atol=tol):
+        raise ValueError("start and end are antipodes"
+                         " using the specified tolerance or they"
+                         " are not on a unit n-sphere")
+
+    # a diameter > 2 within tol violates requirement for input to be on
+    # unit n-sphere
+    if coord_dist > (2.0 + tol):
+        raise ValueError("start and end are not"
+                         " on a unit n-sphere")
+
+    # similarly for points that violate equation for n-sphere
+    # NOTE: this is tricky to think about for the 0-sphere;
+    # performing the check for 1-sphere (circle) and up
+    if start.size > 1:
+        for coord in [start, end]:
+            if not np.allclose(np.square(coord).sum(), 1.0, rtol=0, atol=tol):
```

Comment:
I think there is no reason not to use `np.linalg.norm` --- it does a stable computation and it doesn't create a bottleneck.

As for the tolerance --- a proper normalized array must have norm equal to 1.0 within machine epsilon. So I would consider using `atol=0, rtol=10 * EPS` (for example).

Reply:
I'm using `np.linalg.norm` now---the tolerances were adjusted to allow the tests to continue passing; `rtol` couldn't be as close to double precision limit as I might like, but maybe good enough